Euphorolog.Services/Services/StoriesService.cs

```diff
@@ -16,46 +22,167 @@ namespace Euphorolog.Services.Services
+             * If the story is their's
+             */
+
+            // user logged in
+            var username = _httpContextAccessor.HttpContext?.User?.FindFirst(ClaimTypes.Name)?.Value;
+            if (username == null)
+            {
+                throw new UnAuthorizedException("LogIn first!");
+            }
+
+            // token valid based on time
+            var _utils = new Utilities(_configuration, _httpContextAccessor);
+            if (!_utils.tokenStillValid(await _usersRepository.GetPasswordChangedAtAsync(username)))
+            {
+                throw new UnAuthorizedException("LogIn Again!");
+            }
+
+            //story Exists
+            var story = await _storiesRepository.GetStoryByIdAsync(id);
+            if(story == null)
+            {
+                throw new NotFoundException("Story not Found!");
+            }
+
+            // Their Story
+            if (!String.Equals(story.authorName, username))
+            {
+                throw new ForbiddenException("Not your Story!");
+            }
+            await _storiesRepository.DeleteStoryAsync(id);
+            return;
         }
-        public async Task<Stories> UpdateStoryAsync(string id, Stories story)
+        public async Task<GetStoryByIdResponseDTO> UpdateStoryAsync(string id, UpdateStoryRequestDTO req)
         {
-            return await _storiesRepository.UpdateStoryAsync(id, story);
+            /*
+             * If the user is logged in
+             * If the token is valid based on time
+             * If the story Exists
+             * If the story is their's
+             */
+
+            // user logged in
+            var username = _httpContextAccessor.HttpContext?.User?.FindFirst(ClaimTypes.Name)?.Value;
+            if (username == null)
+            {
+                throw new UnAuthorizedException("LogIn first!");
+            }
+
+            // token valid based on time
+            var _utils = new Utilities(_configuration, _httpContextAccessor);
+            if (!_utils.tokenStillValid(await _usersRepository.GetPasswordChangedAtAsync(username)))
+            {
+                throw new UnAuthorizedException("LogIn Again!");
+            }
+
+            //story Exists
+            var story = await _storiesRepository.GetStoryByIdAsync(id);
+            if (story == null)
+            {
+                throw new NotFoundException("Story not found");
+            }
+
+            // Their Story
+            if (!String.Equals(story.authorName, username))
+            {
+                throw new ForbiddenException("Not your Story!");
+            }
+
+
+            var changedStory = _mapper.Map<Stories>(req);
+            if (changedStory.storyDescription != null)
+            {
+                int openingLineLength = Math.Min(99, changedStory.storyDescription.Length);
+                changedStory.openingLines = changedStory.storyDescription.Substring(0, openingLineLength) + "...";
```

Comment:
IMO: It's better if number of opening lines varies depends on view. I don't recommend it to be in DB